src/util/scoped_timer.cpp

```diff
@@ -33,42 +33,50 @@ Revision History:
 #include <pthread.h>
 #endif
 
+enum scoped_timer_work_state { EXPIRED = -1, IDLE = 0, WORKING = 1, EXITING = 2 };
+
 struct scoped_timer_state {
     std::thread m_thread;
     std::timed_mutex m_mutex;
     event_handler * eh;
     unsigned ms;
-    std::atomic<int> work;
+    std::atomic<scoped_timer_work_state> work;
     std::condition_variable_any cv;
 };
 
 static std::vector<scoped_timer_state*> available_workers;
 static std::mutex workers;
 static atomic<unsigned> num_workers(0);
+static scoped_timer_work_state working = WORKING;
 
 static void thread_func(scoped_timer_state *s) {
     workers.lock();
     while (true) {
-        s->cv.wait(workers, [=]{ return s->work > 0; });
+    start:
+        s->cv.wait(workers, [=]{ return s->work > IDLE; });
         workers.unlock();
 
         // exiting..
-        if (s->work == 2)
+        if (s->work == EXITING)
```

Comment:
@nunoplopes you have got to admit that using enums is so much better than those non-descriptive ints.